Footer company variable

AlexanderBlanchardAC opened the pull request:
Allows the company name in the footer to be set in system variables. Defaults to ByWater Solutions if not changed. 

Test Plan:
1. Scroll to the footer on any of the pages on an Aspen site and note that it says supported by ByWater Solutions.
2. Navigate to Administration and then to Database Maintenance and complete updates. 
3. Navigate to System Variables within Administration and click edit. 
4. The last box should now be an option to edit the name of the supporting company. 
5. Edit the company name if needed and save the changes.
6. Ensure that the footer reflects the changes.

AlexanderBlanchardAC committed:
feat: Adds system variable to allow the supporting company in footer to be changed

-Added system variable, 'supportingCompany' to the system variable table.
-The supporting company in the footer can now be altered by accessing the admin area, then system variables.
-The default is still ByWater Solutions.

Jacobomara901 commented on install/createSite.php:
I think this is code left over from our previous attempts at the solution as this code prompts for the variable and then does nothing with it

Jacobomara901 replied:
I think we can either remove/ implement this depending. Worth a chat maybe?

Jacobomara901 commented on code/web/sys/LibraryLocation/Library.php:
This file we haven't edited so can probably be removed from the commit as its just 2 blank lines.

Jacobomara901 commented on code/web/interface/themes/responsive/footer_responsive.tpl:
```diff
@@ -3,7 +3,9 @@
 		<div class="row {if !empty($fullWidthTheme)}row-no-gutters{/if}">
 			<div class="col-tn-12 col-sm-5 col-md-4 text-left" id="install-info">
-				<small>{translate text='Powered By Aspen Discovery supported by ByWater Solutions' isPublicFacing=true}</small><br>
+		<small>{translate text='Powered By Aspen Discovery supported by ' isPublicFacing=true}</small><small> {$company}</small><br>
```
I think we may be adding an extra space in here between "supported by" and the company name

Also, is the extra small required or can we do:
<small>{translate text='Powered By Aspen Discovery supported by $company' isPublicFacing=true}</small><br>

I think if it works then it look a little cleaner? (this is probably just personal preference though so feel free to push back!)

mrenvoize replied:
Can you put a variable inside a translate block?

AlexanderBlanchardAC replied:
It can go after the translate but within 'small.' Looks neater.

Jacobomara901 commented on code/web/index.php:
```diff
@@ -234,6 +236,17 @@
 
+//system varaible for company name
+$interface->assign('company', 'ByWater Solutions');
+	try {
```
This looks good! I would probably move the indenting back from line 241 onwards by one tab.

Jacobomara901 commented on code/web/index.php:
```diff
@@ -170,6 +170,8 @@
 $productionServer = $configArray['Site']['isProduction'];
 $interface->assign('productionServer', $productionServer);
 
+
```
I would maybe remove the extra lines we have added here?

Jacobomara901 commented on install/createSiteTemplate.ini:
This is also left over from our previous idea I think!

Jacobomara901 commented on install/aspen.sql:
This can be removed as it does not need to be completed. Mark has a script that builds this file based on his current database so manual updates are unnecessary!

Jacobomara901 reviewed (commented):
This looks really good mate, I've left a couple of comments, many of which are probably personal preference! We can discuss how you want to proceed re: the install script over a huddle if you would like?

mrenvoize commented on code/web/interface/themes/responsive/footer_responsive.tpl:
```diff
@@ -3,7 +3,7 @@
 		<div class="row {if !empty($fullWidthTheme)}row-no-gutters{/if}">
 			<div class="col-tn-12 col-sm-5 col-md-4 text-left" id="install-info">
-				<small>{translate text='Powered By Aspen Discovery supported by ByWater Solutions' isPublicFacing=true}</small><br>
+		<small>{translate text='Powered By Aspen Discovery supported by' isPublicFacing=true} {$supportingCompany}</small><br>
```
Very minor thing.. looks like we introduce an unintended leading whitespace change here.. perhaps a switch from tabs to spaces or something in your editor?

mrenvoize commented on code/web/sys/LibraryLocation/Library.php:
```diff
@@ -59,7 +59,6 @@ class Library extends DataObject {
 	public $footerText;
 	public $systemMessage;
-
 	public $generateSitemap;
```
May be worth reverting this line removal.. Whilst I agree, it cleans the file up a little.. I'm not sure if it's worth touching this file as part of this overall patch set.

Rather nitpicky.. Apologies.

mrenvoize reviewed (commented):
This looks great.. tiny comments, but just about neatness rather than actual functional issues.

I imagine we could include a database file at install, me to run via Ansible to get this set to PTFS Europe for our installations to be default at install time.  One to catch @rhen976 for once this is merged up.

AlexanderBlanchardAC committed:
docs: Updated patch notes for system variables. 'Supporting Company' addition to variables.

Updated the 23.07.00 patch notes to outline the addition of a system variable to allow the supporting company name to be edited.

AlexanderBlanchardAC committed:
Merge branch 'footer_company_variable' into footer_company_variable
